api/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -170,6 +170,88 @@ type SingleStepInstrumentation struct {
+// can be converted to a metav1.LabelSelector using the AsLabelSelector method. Full config key:
+// apm_config.instrumentation.targets[].selector
+type PodSelector struct {
+	// MatchLabels is a map of key-value pairs to match the labels of the pod. The labels and expressions are ANDed.
+	// Full config key: apm_config.instrumentation.targets[].podSelector.matchLabels
+	MatchLabels map[string]string `json:"matchLabels"`
+	// MatchExpressions is a list of label selector requirements to match the labels of the pod. The labels and
+	// expressions are ANDed. Full config key: apm_config.instrumentation.targets[].podSelector.matchExpressions
+	MatchExpressions []SelectorMatchExpression `json:"matchExpressions"`
+}
+
+// SelectorMatchExpression is a reconstruction of the metav1.LabelSelectorRequirement struct to be able to unmarshal
+// the configuration. Full config key: apm_config.instrumentation.targets[].selector.matchExpressions
+type SelectorMatchExpression struct {
+	// Key is the key of the label to match. Full config key:
+	// apm_config.instrumentation.targets[].selector.matchExpressions[].key
+	Key string `json:"key"`
+	// Operator is the operator to use to match the label. Valid values are In, NotIn, Exists, DoesNotExist.
+	Operator metav1.LabelSelectorOperator `json:"operator"`
+	// Values is a list of values to match the label against. If the operator is Exists or DoesNotExist, the values
+	// should be empty. If the operator is In or NotIn, the values should be non-empty.
+	Values []string `json:"values"`
+}
+
+// NamespaceSelector is a struct to store the configuration for the namespace selector. It can be used to match the
+// namespaces to apply the auto instrumentation to. Full config key:
+// apm_config.instrumentation.targets[].namespaceSelector
+type NamespaceSelector struct {
+	// MatchNames is a list of namespace names to match. If empty, all namespaces are matched. Full config key:
+	// apm_config.instrumentation.targets[].namespaceSelector.matchNames
+	MatchNames []string `json:"matchNames"`
+	// MatchLabels is a map of key-value pairs to match the labels of the namespace. The labels and expressions are
+	// ANDed. This cannot be used with MatchNames. Full config key:
+	// apm_config.instrumentation.targets[].namespaceSelector.matchLabels
+	MatchLabels map[string]string `json:"matchLabels"`
+	// MatchExpressions is a list of label selector requirements to match the labels of the namespace. The labels and
+	// expressions are ANDed. This cannot be used with MatchNames. Full config key:
+	// apm_config.instrumentation.targets[].selector.matchExpressions
+	MatchExpressions []SelectorMatchExpression `json:"matchExpressions"`
+}
+
+// TracerConfig is a struct that stores configuration options for a tracer. These will be injected as environment
+// variables to the workload that matches targeting.
+type TracerConfig struct {
```

Comment:
`[]corev1.EnvVar` could be used instead

Reply:
We've used our own types top be able to control the struct tags attached. These also are used in the agent here, and we're trying to keep the types mirrored: https://github.com/DataDog/datadog-agent/blob/87919c66953aac6457eccc55af944684b41be074/pkg/clusteragent/admission/mutate/autoinstrumentation/config.go#L206